src/module/migration/migrations/817-field-discovery-predicates.ts

```diff
@@ -0,0 +1,37 @@
+import { ItemSourcePF2e } from "@item/data";
+import { AELikeSource } from "@module/rules/rule-element/ae-like";
+import { MigrationBase } from "../base";
+
+/** Convert crafting-entry field discovery data to predicates */
+export class Migration817FieldDiscoveryPredicates extends MigrationBase {
+    static override version = 0.817;
+
+    override async updateItem(source: ItemSourcePF2e): Promise<void> {
+        const aeLikes = source.system.rules.filter(
+            (r): r is AELikeSource =>
+                r.key === "ActiveEffectLike" &&
+                "path" in r &&
+                r.path === "system.crafting.entries.alchemist.fieldDiscovery"
```

Comment:
I think if you wanna try handling homebrew, you could try matching on entries other than alchemist. The CraftingEntry class doesn't restrict it to just alchemist.

Reply:
Updated